### PR TITLE
responsive navigation header and side drawer

### PR DESCRIPTION
**Pull request by balveda**
Created new pages: pages/about, pages/contact
Added routes to src/routes.tsx
Added linkText attribute to routes objects to display in app header and side drawer
Added showInHeader attribute to routes objects to distinguish routes that might only be showed in footer etc
Added links (pulled from routes array) and a user menu to appHeader
Added side drawer for showing links on mobile

/**
 * Opening the user menu and side drawer generates a warning in the console
 * Apparently this is a known issue when running material ui in strict mode https://github.com/mui-org/material-ui/issues/13394
 * From my reading this will not happen in production and can be resolved by removing strict mode

 */

### CODE REVIEW

File: src/layout/appHeader.tsx [hunk 1]
```diff
@@ -1,34 +1,159 @@
...
+
+/**
+ * Opening the user menu and side drawer generates a warning in the console
+ * Apparently this is a known issue when running material ui in strict mode https://github.com/mui-org/material-ui/issues/13394
+ * From my reading this will not happen in production and can be resolved by removing strict mode
+ */
+
+export const AppHeader: FC<Props> = ({sideDrawOpen, setSideDrawerOpen}) => {
+  const {isAuthenticated, loginWithPopup, logout, user} = useAuth0()
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
```
Comment (dhargitai) [inline src/layout/appHeader.tsx]: I think we shouldn't use contraction anywhere, as it contributes only to harder readability.

**Comment by dhargitai** [comment]
Apart from that tiny little thing (contraction) it looks good to me, Tom. 👍

File: src/layout/appHeader.tsx (hunk 1 above)
Comment (balveda) [reply src/layout/appHeader.tsx]: Yes, I agree Dave. I'll fix it. Thanks.

File: src/layout/appHeader.tsx [hunk 2]
```diff
@@ -1,34 +1,159 @@
...
+import MenuIcon from '@material-ui/icons/Menu'
 import {useAuth0} from '@auth0/auth0-react'
+import {routes} from '../routes'
 
-export const AppHeader: FC = props => {
-  const {isAuthenticated, loginWithPopup, logout} = useAuth0()
+interface Props {
+  sideDrawOpen: boolean
+  setSideDrawerOpen: Dispatch<SetStateAction<boolean>>
+}
```
Comment (marcosignorello) [inline src/layout/appHeader.tsx]: I would rename 
sideDrawOpen to isSideDrawOpen
and 
setSideDrawerOpen to openSideDrawer

so it's self explanatory

File: src/layout/appHeader.tsx [hunk 3]
```diff
@@ -1,34 +1,159 @@
...
+    menuButton: {
+      marginLeft: theme.spacing(1),
+    },
+    title: {
+      flexGrow: 1,
+    },
+    navLink: {
+      padding: theme.spacing(1),
+    },
+  }))
```
Comment (marcosignorello) [inline src/layout/appHeader.tsx]: What do you guys think if we move this separate file in case we see it grow?

File: src/layout/sideDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  const classes = useStyles()
+
+  const toggleDrawer = (open: boolean) => (
+    event: React.KeyboardEvent | React.MouseEvent,
+  ) => {
+    if (
+      event &&
+      event.type === 'keydown' &&
+      ((event as React.KeyboardEvent).key === 'Tab' ||
+        (event as React.KeyboardEvent).key === 'Shift')
```
Comment (marcosignorello) [inline src/layout/sideDrawer.tsx]: I would deal with this event in a function that accept an event and return a boolean

File: src/layout/appHeader.tsx (hunk 1 above)
Comment (marcosignorello) [reply src/layout/appHeader.tsx]: Great work, Tom! I have just added a couple of comments.

Comment (balveda) [reply src/layout/appHeader.tsx]: I have changed variable/class names in appHeader.tsx, sideDrawer.tsx, and layout.tsx to improve readability.

File: src/layout/appHeader.tsx (hunk 2 above)
Comment (balveda) [reply src/layout/appHeader.tsx]: Thanks Marco. I changed the names to isSideDrawerOpen and setIsSideDrawerOpen. Much clearer now I hope.

File: src/layout/sideDrawer.tsx (hunk 4 above)
Comment (balveda) [reply src/layout/sideDrawer.tsx]: Thanks very much for this Marco. Revisiting my code I realised that I had just copy/pasted from the material ui docs without completely understanding the code. The code you identified is not necessary and was performing a specific function within the docs demo that is not necessary in our context. We don't need access to the event which has greatly simplified the code.

File: src/layout/appHeader.tsx (hunk 3 above)
Comment (balveda) [reply src/layout/appHeader.tsx]: My feeling is that these styles are very specific to the app header and probably won't be used elsewhere within the code base. I am sure that we will use a separate styles file as the project grows and I would be happy to move these to there if required. I would need to make the class names a lot more specific though.

**Review by marcosignorello — APPROVED** [review]

**Pull request merged**